### PR TITLE
feat(AWS): Lightsail new service and checks

### LINKED ISSUE #3869: Add support for AWS Lightsail resource
### New feature motivation

It would be great if Prowler starts supporting AWS Lightsail as this as well is a common service used by several teams.

### Solution Proposed

Support AWS Lightsail instance scanning for things like:
- Open ports
- Static IPs
- Snapshots
- Disk usage

### Describe alternatives you've considered

NA

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by puchy22**
### Context

Include new AWS service `Lightsail` with checks:

- [x] lightsail_instance_public
- [x] lightsail_database_public
- [x] lightsail_instance_automated_snapshots
- [x] lightsail_static_ip_unused

### Description

This PR add new services, checks and respective unit tests.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/lightsail/lightsail_instance_public/lightsail_instance_public.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            report.resource_arn = arn_instance
+            report.resource_tags = instance.tags
+            report.status = "FAIL"
+            report.status_extended = f"Instance '{instance.name}' has public access"
+            if instance.public_ip == "" and not any(
+                port.access_type == "public" for port in instance.ports
+            ):
+                report.status = "PASS"
+                report.status_extended = (
+                    f"Instance '{instance.name}' has no public access"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/lightsail/lightsail_instance_public/lightsail_instance_public.py]:
```suggestion
                    f"Instance '{instance.name}' is not publicly exposed."
```

File: prowler/providers/aws/services/lightsail/lightsail_instance_public/lightsail_instance_public.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        findings = []
+
+        for arn_instance, instance in lightsail_client.instances.items():
+            report = Check_Report_AWS(self.metadata())
+            report.region = instance.region
+            report.resource_id = instance.id
+            report.resource_arn = arn_instance
+            report.resource_tags = instance.tags
+            report.status = "FAIL"
+            report.status_extended = f"Instance '{instance.name}' has public access"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/lightsail/lightsail_instance_public/lightsail_instance_public.py]: Can you put the ports that are opened?

File: prowler/providers/aws/services/lightsail/lightsail_instance_public/lightsail_instance_public.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            if instance.public_ip == "" and not any(
+                port.access_type == "public" for port in instance.ports
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/lightsail/lightsail_instance_public/lightsail_instance_public.py]: I cannot see the attribute `access_type` in the Port class

File: prowler/providers/aws/services/lightsail/lightsail_static_ip_unused/lightsail_static_ip_unused.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            report.resource_tags = []
+            report.status = "FAIL"
+            report.status_extended = (
+                f"Static IP '{static_ip.name}' is not associated with any instance."
+            )
+
+            if static_ip.is_attached:
+                report.status = "PASS"
+                report.status_extended = (
+                    f"Static IP '{static_ip.name}' is associated with any instance."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/lightsail/lightsail_static_ip_unused/lightsail_static_ip_unused.py]: Can you put the instance that is attached to?

File: prowler/providers/aws/services/lightsail/lightsail_instance_public/lightsail_instance_public.py (hunk 3 above)
Comment (puchy22) [reply prowler/providers/aws/services/lightsail/lightsail_instance_public/lightsail_instance_public.py]: https://github.com/prowler-cloud/prowler/pull/3919/files#diff-1ccedd3473e8d0ba377e274a6434e22ecfddb9cc831f2c1e3aa0996a3a90b2beR165

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**